scalafmt-cli/shared/src/main/scala/org/scalafmt/cli/ScalafmtCoreRunner.scala

```diff
@@ -31,41 +27,29 @@ object ScalafmtCoreRunner extends ScalafmtRunner {
       try ProjectFiles.FileMatcher(scalafmtConf.project, options.customExcludes)
       catch {
         case e: ScalafmtConfigException =>
-          options.common.err.println(e.getMessage())
+          options.common.err.println(e.getMessage)
           return ExitCode.UnexpectedError // RETURNING EARLY!
       }
 
     val inputMethods = getInputMethods(options, filterMatcher.matchesPath)
-    val adjustedScalafmtConf = {
-      if (scalafmtConf.needGitAutoCRLF) options.gitOps.getAutoCRLF else None
-    }.fold(scalafmtConf)(scalafmtConf.withGitAutoCRLF)
+    if (inputMethods.isEmpty) ExitCode.Ok
+    else {
+      val adjustedScalafmtConf = {
+        if (scalafmtConf.needGitAutoCRLF) options.gitOps.getAutoCRLF else None
+      }.fold(scalafmtConf)(scalafmtConf.withGitAutoCRLF)
 
-    val termDisplay = newTermDisplay(options, inputMethods, termDisplayMessage)
-    val exitCode = new AtomicReference(ExitCode.Ok)
-    Breaks.breakable(inputMethods.compatPar.foreach { inputMethod =>
-      val code = handleFile(inputMethod, options, adjustedScalafmtConf)
-      exitCode.getAndUpdate(ExitCode.merge(code, _))
-      if (options.check && !code.isOk) Breaks.break
-      termDisplay.taskProgress(termDisplayMessage)
-    })
-    termDisplay.completedTask(termDisplayMessage, exitCode.get.isOk)
-    termDisplay.stop()
-    exitCode.get()
+      runInputs(options, inputMethods, termDisplayMessage) { inputMethod =>
+        import org.scalafmt.sysops.PlatformCompat.executionContext
```

Comment:
Curiously, do we also enable multithreading in Scala Native as well this way? Maybe we don't need the separate executionContext anymore?

Maybe we can remove compatPar now also?

Reply:
> during building i see messages about multithreading enabled, perhaps we could somehow make it explicit

I think it's automatically disabled when not needed, so no need to make it explicit I guess?

> i can't see much difference in speed in community test running, but it felt to me much faster when i ran the result of sbt cli/assembly with local snapshot version (forcing core runner, instead of dynamic).

Och, interesting! Do you mean the JVM version?

Reply:
yes, the JVM. native was even faster but i didn't check how native performed without this change.